### PR TITLE
Update url-set-builder.ts

### PR DESCRIPTION
**Pull request by tomelliot**
Add support for route groups

### CODE REVIEW

**Comment by djanowski** [comment]
@iamvishnusankar 👋  Any chance we can get this merged?

**Comment by tomelliot** [comment]
> 👋 Any chance we can get this merged?

@djanowski FYI I coincidentally pinged @iamvishnusankar on Twitter a couple of days ago - he said he'd get to it in a few days 🤞

**Comment by iamvishnusankar** [comment]
@tomelliot Thanks for the PR. Seems like some tests are failing!

**Comment by tomelliot** [comment]
@iamvishnusankar can you provide some info on how to setup the tests locally? I'm trying to run the tests for the current master commit (so I can then fix the tests that are broken in my PR). I'm getting failing results for the current master commit - obviously I haven't got things set up correctly.

I also noticed that the tests have moved from jest to bun, and bun isn't marked as a dev dependency in package.json (I guess it should be?).

Example test output
[next-sitemap-test.txt](https://github.com/iamvishnusankar/next-sitemap/files/13672535/next-sitemap-test.txt)

File: packages/next-sitemap/src/interface.ts [hunk 1]
```diff
@@ -196,7 +196,13 @@ export interface IRoutesManifest {
...
     locales: string[]
     defaultLocale: string
-  }
+  },
+  dynamicRoutes: [{
+    [key: string]: string[]
+  }],
+  staticRoutes: [{
+    [key: string]: string[]
+  }]
```
Comment (jlubawy) [inline packages/next-sitemap/src/interface.ts:205]: This seems to fix the build errors for me at least:
```suggestion
  dynamicRoutes?: {
    page: string
  }[]
  staticRoutes?: {
    page: string
  }[]
```

**Pull request closed without merging**